testbed/testbed/senders.go

```diff
@@ -65,26 +65,48 @@ type MetricDataSender interface {
 	SendMetrics(metrics consumerdata.MetricsData) error
 }
 
-// JaegerDataSender implements TraceDataSender for Jaeger Thrift-HTTP protocol.
-type JaegerDataSender struct {
-	// exporter *jaeger.Exporter
+// DataSenderOverTraceExporter partially implements TraceDataSender via a TraceExporter.
+type DataSenderOverTraceExporter struct {
 	exporter exporter.TraceExporter
-	port     int
+	Port     int
+}
+
+// NewDataSenderOverExporter creates a new sender that will send
+// to the specified port after Start is called.
+func NewDataSenderOverExporter(port int) *DataSenderOverTraceExporter {
+	return &DataSenderOverTraceExporter{Port: port}
+}
+
+func (ds *DataSenderOverTraceExporter) SendSpans(traces consumerdata.TraceData) error {
+	return ds.exporter.ConsumeTraceData(context.Background(), traces)
+}
+
+func (ds *DataSenderOverTraceExporter) Flush() {
+	// TraceExporter interface does not support Flush, so nothing to do.
```

Comment:
nit: related I usually like to have the `var _ ExpectedInterface = ...` to make clear what are the required interfaces, even when the usage allows the compiler to catch it.